zulipterminal/model.py

```diff
@@ -110,17 +110,53 @@ def __init__(self, controller: Any) -> None:
 
         subscriptions = self.initial_data['subscriptions']
         stream_data = Model._stream_info_from_subscriptions(subscriptions)
-        (self.stream_dict, self.muted_streams,
+        (self.stream_dict, (self.muted_streams, self.initial_unmuted_streams),
          self.pinned_streams, self.unpinned_streams) = stream_data
 
         self.muted_topics = self.initial_data['muted_topics']
+
+        # Fetch additional messages to ensure non-empty all-messages narrow
+        # (caused by muted streams)
+        messages_not_muted = [
+            message
+            for message in self.index['messages'].values()
+            if (message['type'] == 'stream' and
+                message['stream_id'] not in self.muted_streams and
+                ([message['display_recipient'], message['subject']]
+                 not in self.muted_topics))
+        ]
+        if len(messages_not_muted) == 0:
+            self.__fetch_additional_initial_messages(workers=1)
+
         self.unread_counts = classify_unread_counts(self)
 
         self.fetch_all_topics(workers=5)
 
         self.new_user_input = True
         self._start_presence_updates()
 
+    def __fetch_additional_initial_messages(self, *, workers: int) -> None:
+        assert workers > 0
+        unmuted_stream_narrows = [
+            [['stream', self.stream_dict[stream_id]['name']]]
+            for stream_id in self.initial_unmuted_streams
+        ]
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            futures = {
```

Comment:
@neiljp did you profile this? I think fetching messages for all the unmuted_stream_narrows will take a lot of time. (Assume some realm has like 100 unmuted streams)